src/utils/CrossOriginStorage.js

```diff
@@ -0,0 +1,71 @@
+class CrossOriginStorage {
+  static isAvailable = () => "crossOriginStorage" in navigator;
+
+  match = async (request) => {
+    const hashValue = await this._getFileHash(request);
+    if (!hashValue) {
+      return undefined;
+    }
+    const hash = { algorithm: "SHA-256", value: hashValue };
```

Comment:
I know I hard-coded it, but future versions of COS may use other hashing algorithms. So to future-proof this, maybe make this stand out more by putting it in a constant at the top.